.github/workflows/build-push.yml

```diff
@@ -0,0 +1,335 @@
+            echo "You have two options:"
+            echo ""
+            echo "1. Use a 'dev/' prefixed branch (e.g., 'dev/main', 'dev/feature-branch')"
+            echo "   - Source: dev/main → Build artifacts: main"
+            echo "   - Source: dev/ABC-123 → Build artifacts: ABC-123"
+            echo ""
+            echo "2. Provide an explicit BUILT_BRANCH_NAME input to specify where build artifacts should go"
+            echo ""
+            echo "Current branch: $ORIGIN_BRANCH"
+            echo "Target build branch: ${{ env.BUILT_BRANCH_NAME }}"
+            echo ""
+            exit 1
+          fi
+          
+          echo "✅ Branch strategy validated:"
+          echo "  Source branch: $ORIGIN_BRANCH"
+          echo "  Build branch: ${{ env.BUILT_BRANCH_NAME }}"
+
+      - name: Checkout the build branch and merge in it
+        run: |
+          git show-ref -q refs/remotes/origin/${{ env.BUILT_BRANCH_NAME }} && git checkout ${{ env.BUILT_BRANCH_NAME }} || git checkout -b ${{ env.BUILT_BRANCH_NAME }}
+          git checkout ${{ github.ref_name }} -- .
+
+      - name: Set up PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          php-version: ${{ inputs.PHP_VERSION }}
+          tools: php-scoper, sniccowp/php-scoper-wordpress-excludes, composer/installers, inpsyde/wp-translation-downloader, ${{ inputs.PHP_TOOLS }}
+          coverage: none
+
+      - name: Install Composer dependencies
+        uses: ramsey/composer-install@v3
+        env:
+          COMPOSER_AUTH: '${{ secrets.COMPOSER_AUTH_JSON }}'
+        with:
+          composer-options: ${{ inputs.COMPOSER_ARGS }}
+
+      - name: Set up node
+        uses: actions/setup-node@v4
+        env:
+          NODE_OPTIONS: ${{ inputs.NODE_OPTIONS }}
+          NODE_AUTH_TOKEN: ${{ secrets.NPM_REGISTRY_TOKEN }}
+        with:
+          node-version: ${{ inputs.NODE_VERSION }}
+          registry-url: ${{ inputs.NPM_REGISTRY_DOMAIN }}
+          cache: npm
+
+      - name: Install dependencies
+        run: npm ci
+
+      - name: Determine package type
+        run: |
+          if [ -f "style.css" ]; then
+            echo "Package type: WordPress Theme"
+            echo "IS_THEME=true" >> $GITHUB_ENV
+          else
+            echo "Package type: WordPress Plugin"
+            echo "IS_THEME=false" >> $GITHUB_ENV
+          fi
+
+      - name: Prepare Plugin
+        if: ${{ env.IS_THEME == 'false' }}
+        run: |
+          # Find and update plugin main file
+          PLUGIN_FILE=$(grep -rl --include "*.php" "Plugin Name:" . | head -1)
+          if [ -n "$PLUGIN_FILE" ]; then
+            echo "Updating plugin file: $PLUGIN_FILE"
+            sed -Ei "s/Version: .*/Version: ${{ env.PACKAGE_VERSION }}/g" "$PLUGIN_FILE"
+            sed -Ei "s/SHA: .*/SHA: ${{ github.sha }}/g" "$PLUGIN_FILE"
+          else
+            echo "Warning: No plugin main file found!"
+          fi
+
+      - name: Prepare Theme
```

Comment:
While I think it is good to have a check for the Plugins Headers and Theme style.css Headers, should we maybe also support something dynamic like a global constant, which could also contain the version? This could become handy for libraries that do not have a version set anywhere in code and are parseable through WordPress.

Reply:
We have some libraries (not Plugins) which also contain a version that is shown in the Backend, or, depending on the version, something needs to be triggered (like updating the DB tables). In this case, a version string also needs to be added and updated manually somewhere in a file. 
Through `inpsyde/modularity`, we only have the possibility right now to set `version` in the `composer.json` for libraries - see https://github.com/inpsyde/modularity/blob/master/src/Properties/LibraryProperties.php#L89-L93

My thought here was to also cover the need to have a "version" changed in code for "PHP libraries".

Reply:
https://github.com/inpsyde/reusable-workflows/pull/194/commits/91021336700af3e9011e019bf2181dd883d2289b